### PR TITLE
Feat/add mockall

### PR DESCRIPTION
**Pull request by dorezyuk**
Adds mocks to the project to allow simpler unit testing in user code

### CODE REVIEW

File: everestrs/everestrs/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,8 @@ argh = "0.1.10"
 cxx = { version = "1.0.107", features = ["c++17"] }
 everestrs-build = { workspace = true }
 log = "0.4.20"
+mockall = "0.12.1"
+mockall_double = "0.3.1"
```
Comment (golovasteek) [inline everestrs/everestrs/Cargo.toml:12]: Shouldn't it be dev dependency?

Comment (dorezyuk) [reply everestrs/everestrs/Cargo.toml:12]: ideally yes but not possible

**Comment by SirVer** [comment]
Should this be behind a feature gate? I am not sure how heavy a dependency mockall is? Otherwise lgtm.

File: everestrs/everestrs/Cargo.toml (hunk 1 above)
Comment (golovasteek) [reply everestrs/everestrs/Cargo.toml:12]: We should remove it now right?

**Review by golovasteek — APPROVED** [review]

**Pull request merged**